src/main/java/com/xceptance/neodymium/util/SelenideAddons.java

```diff
@@ -211,47 +213,39 @@ public ElementsCollection get()
      */
     public static void $safe(final Runnable code)
     {
-        final int maxRetryCount = Neodymium.configuration().staleElementRetryCount();
-        int retryCounter = 0;
-
-        while (retryCounter <= maxRetryCount)
-        {
-            try
-            {
-                code.run();
-                break;
-            }
-            catch (final Throwable t)
-            {
-                if (isThrowableCausedBy(t, StaleElementReferenceException.class))
-                {
-                    retryCounter++;
-                    if (retryCounter > maxRetryCount)
-                    {
-                        // fail
-                        throw t;
-                    }
-                    else
-                    {
-                        AllureAddons.addToReport("StaleElementReferenceException catched times: \"" + retryCounter + "\".", retryCounter);
-                        Selenide.sleep(Neodymium.configuration().staleElementRetryTimeout());
-                    }
-                }
-                else
-                {
-                    // not the kind of error we are looking for
-                    throw t;
-                }
-            }
-        }
+        $safe(() -> {
+            code.run();
+            return null;
+        });
     }
 
-    private static boolean isThrowableCausedBy(final Throwable throwable, Class<? extends Throwable> clazz)
+    /**
+     * Recursively checks if the throwable is caused by exception of the specific class or contains one of the messages
+     * listed in the {@code phrasesHintingErrorToCatch}
+     * 
+     * @param throwable
+     *            throwable to check
+     * @param clazz
+     *            class of the expected throwable cause
+     * @param phrasesHintingErrorToCatch
+     *            optional parameters, error messages of expected throwable or its causes
+     * @return result of the check as boolean value
+     */
+    public static boolean isThrowableCausedBy(final Throwable throwable, Class<? extends Throwable> clazz, String... phrasesHintingErrorToCatch)
     {
         Throwable t = throwable;
         while (t != null)
         {
-            if (clazz.isInstance(t))
+            boolean containsMessage = false;
+            for (String message : phrasesHintingErrorToCatch)
+            {
+                if (t.getMessage().contains(message))
```

Comment:
Better version. More readable and with null check
```
 for (String phrase : phrasesHintingErrorToCatch)
            {
                String message  = t.getMessage();
                if (message !=null && message.contains(phrase))
                {
```

Reply:
Done